core/src/repair/repair_service.rs

```diff
@@ -612,26 +621,29 @@ impl RepairService {
         slot: Slot,
         slot_meta: &SlotMeta,
         max_repairs: usize,
+        add_delay_after_first_shred: bool,
     ) -> Vec<ShredRepairType> {
         if max_repairs == 0 || slot_meta.is_full() {
             vec![]
         } else if slot_meta.consumed == slot_meta.received {
             // check delay time of last shred
-            if let Some(reference_tick) = slot_meta
-                .received
-                .checked_sub(1)
-                .and_then(|index| blockstore.get_data_shred(slot, index).ok()?)
-                .and_then(|shred| shred::layout::get_reference_tick(&shred).ok())
-                .map(u64::from)
-            {
-                // System time is not monotonic
-                let ticks_since_first_insert = DEFAULT_TICKS_PER_SECOND
-                    * timestamp().saturating_sub(slot_meta.first_shred_timestamp)
-                    / 1_000;
-                if ticks_since_first_insert
-                    < reference_tick.saturating_add(DEFER_REPAIR_THRESHOLD_TICKS)
+            if add_delay_after_first_shred {
```

Comment:
you should just pass in `defer_repair_threshold_ticks` instead of the boolean here, you can set it to 0 for `wen_restart`, since you'll need to pass it below to `blockstore.find_missing_data_indexes` as well.

Reply:
Hmm, I tried that, but this would leak DEFER_REPAIR_THRESHOLD_TICKS outside the file. Since the only thing we want to do is turning it off, I kept the boolean variable and changed blockstore.find_missing_data_indexes, does this look better?

Reply:
ya good call, this works!